packages/create-plugin/src/utils/utils.templates.ts

```diff
@@ -23,9 +23,13 @@ export function getTemplateFiles(pluginType: string, filter?: string | string[])
 
   if (filter) {
     return templateFiles.filter((file) => {
-      const projectRelativePath = getProjectRelativeTemplatePath(file, pluginType);
+      const fileProjectRelativePath = getProjectRelativeTemplatePath(file, pluginType);
+      const exportedName = path.join(path.dirname(fileProjectRelativePath), getExportFileName(fileProjectRelativePath));
 
-      return isFileStartingWith(projectRelativePath, filter);
+      // note: some files to filter might use their exported name instead of
+      // their template version e.g. _eslint instead of .eslint.
+      // We need to check the filter with both of them to prevent skipping them in an update or migration
+      return isFileStartingWith(fileProjectRelativePath, filter) || isFileStartingWith(exportedName, filter);
```

Comment:
@jackw my alternative fix to have a double check here is to have an inverse function to `getExportFileName` such as `getTemplateFileName` which would transform the list of files to migrate to their "template" version and then the rest of the code handles it correctly.